Resource adjustments and updated obsproc/prepobs packages for tcvitals bug fix

KateFriedman-NOAA opened the pull request:
**Description**

This PR includes memory updates and new obsproc/prepobs package versions:

- Memory increases for some non-exclusive jobs that were getting the `Cgroup mem limit exceeded` warning message on WCOSS2.
- Also translated a few more high res WCOSS2 resource settings from `config.resources.nco.static` to `config.resources.emc.dyn` and from `config.fv3.nco.static` to `config.fv3.emc.dyn`.
- New obsproc R&D v1.0.2, includes WCOSS2 ops updates and tcvitals bug fix. Updated `obsproc_run_ver` in `versions/*.ver` files for WCOSS2, Hera, Orion.
- New prepobs R&D v1.0.1, includes WCOSS2 ops updates and tcvitals bug fix. Updated `prepobs_run_ver` in `versions/*.ver` files for WCOSS2, Hera, Orion.

New obsproc/prepobs packages have been installed on WCOSS2, Hera, and Orion.

Will merge these changes into the `release/gfs.v16.3.0` branch next.

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**

- [x] Clone and build tests on WCOSS2, Hera, Orion (obsproc/prepobs packages)
- [x] Cycled test on WCOSS2
  
Refs #665, #744

KateFriedman-NOAA committed:
Resource adjustments for memory and dev

- Increase memory request values for some non-exclusive jobs that
have been getting Cgroup mem warning messages on WCOSS2.
- Translate additional high res resource values from
config.resources.nco.static into config.resources.emc.dyn.

Refs: #665, #744

KateFriedman-NOAA committed:
Memory adjustments in ecf scripts

- Adjust/increase memory requests for some non-exclusive jobs that were
getting the Cgroup mem warning messages on WCOSS2.
- Some additional memory adjustments to wave jobs in resource configs.

Refs: #665, #744

KateFriedman-NOAA committed:
Update obsproc and prepobs versions in version files

- Update obsproc_run_ver to 1.0.2-rd
- Update prepobs_run_ver to 1.0.1-rd

New obsproc/prepobs versions include tcvitals bug fix.

Refs #665

aerorahul commented on parm/config/config.resources.emc.dyn:
```diff
@@ -43,7 +43,9 @@ if [ $step = "prep" -o $step = "prepbufr" ]; then
     eval "export npe_node_$step=2"
     eval "export nth_$step=1"
-    eval "export memory_$step=40G"
+    if [[ "$machine" != "WCOSS2" ]]; then
+      eval "export memory_$step=40G"
+    fi
```
What value for memory is being given on WCOSS2?
Not sure if this warrants an `if-block`

KateFriedman-NOAA replied:
The prep job runs exclusively on one node on WCOSS2 so we don't want a memory setting for it in here (setup scripts only set exclhost in the xml if no memory is set for job). The prep job fails with 40G on WCOSS2 when tested, 40G is the R&D value that has been tested.

aerorahul replied:
Ok. thanks.  
In a subsequent PR, we might want to know what the memory footprint is on WCOSS2

KateFriedman-NOAA replied:
Will do, thanks!

aerorahul commented on parm/config/config.resources.emc.dyn:
```diff
@@ -192,15 +195,16 @@ elif [ $step = "fcst" ]; then
     export npe_node_fcst=$(echo "$npe_node_max / $nth_fcst" | bc)
     export npe_node_fcst_gfs=$(echo "$npe_node_max / $nth_fcst_gfs" | bc)
     if [[ "$machine" == "WCOSS2" ]]; then
+       export wtime_fcst_gfs="02:30:00"
```
Is this really necessary?
Or is WCOSS2 really that fast, that `6:00:00` on line 190 a "safe bet"?

KateFriedman-NOAA replied:
The 6hr gfs runtime covers the job for all resolutions on all machines, so yeah, line 190 is the "safe bet". It's not super necessary (beyond the possibility of jobs waiting longer on WCOSS2 down the road when the machine gets busy) so I have removed the line. Can review runtimes again later.

aerorahul commented on parm/config/config.resources.emc.dyn:
```diff
@@ -192,15 +195,16 @@ elif [ $step = "fcst" ]; then
+       export wtime_fcst_gfs="02:30:00"
        export npe_node_fcst=32
-       export npe_node_fcst_gfs=42
+       export npe_node_fcst_gfs=24
```
How do the values on L200 and L199 differ from the calculations on L195 and L196?

KateFriedman-NOAA replied:
L195 = 128/3 = 42.6, which rounds up to 43 in the xml
L196 = 128/5 = 25.6, which rounds up to 26 in the xml

Neither value lays nicely across the WCOSS2 nodes.

George V and I did a lot of testing to get the values on L199 and L200. We want users using these values for C768 on WCOSS2. May also want them for lower resolutions too but haven't fully vetted those resolutions on WCOSS2 yet.

This reminded me I forgot to adjust the C768 block in `config.fv3.emc.dyn` with the tested values for `layout_x_gfs` and `WRTTASK_PER_GROUP_GFS` from `config.fv3.nco.static`. Please see that change in this PR now too.

aerorahul replied:
Ok. So these values are specific to C768.
May be then please add a note to that effect.

KateFriedman-NOAA replied:
Done. I added a check for CASE=768 to make sure those values aren't used outside of high res on WCOSS2.

aerorahul reviewed (commented):
Nothing major, but a few questions.

KateFriedman-NOAA committed:
Update C768 block in emc.dyn with nco.static values

- Update the C768 block in config.fv3.emc.dync with the tested values
from config.fv3.nco.static.
- Will be used on WCOSS2 for C768 but not on R&D machines because of
npe_node_max checks.

Refs: #665

KateFriedman-NOAA committed:
Remove additional wtime_fcst_gfs in fcst block

Refs: #665

aerorahul reviewed (approved):
Looks good.  Just add a note about C768 for those calculations and that they will be revised for other resolutions when we get to it.

KateFriedman-NOAA committed:
Add check of CASE=768 to fcst npe_node_fcst values

- Add a check that CASE=768 in the fcst block of
config.resources.emc.dyn where specific values of npe_node_fcst are set
for WCOSS2.

Refs: #665
